appserver/node-express/lib/db-client/qnaDoc/patch.js

```diff
@@ -66,6 +116,25 @@ module.exports = function (txid, spec) {
     );
   };
 
+  /**
+   * Handle the post of a new answer to a Samplestack question. The posted
+   * answer is merged with an answer template object, a unique ID, contributor
+   * information, and the current time. The resulting object is written to the
+   * MarkLogic database. During the write, the answer count for the question is
+   * incremented and the last activity date is updated.
+   *
+   * @param  {String} txid The transaction ID.
+   * @param  {Object} contributor Contributor object. Example:
+   *   {"id":"cf99542d-f024-4478-a6dc-7e723a51b040",
+   *    "displayName":"JoeUser"}
+   * @param  {String} questionId The question ID.
+   * @param  {Object} spec Answer content with "comments" and "text" properties.
+   * The comments property is an empty array (TODO Why? Remove?). Example:
+   *   {"comments":[],
```

Comment:
Are you saying comments need to be supplied, or we're calling this function with comments that need not necessarily be supplied? I believe what we end up storing is set to match the Java code's behavior. If we don't need to add empty commebnts, then we shouldn't. If we **do** need empty comments, we should add them **here** as that requirement is really a db-client concern (probably related to the indexing) rather than an application-level necessity).